lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -2784,4 +2784,212 @@ bool operator==(const Teids& lhs, const Teids& rhs) {
+          insert_dynamic_5g_rule(dynamic_rule, lifetime, session_uc);
+      pending_activation->push_back(to_process);
+    }
+  }
+}
+
+RuleToProcess SessionState::activate_static_5g_rule(
+    const std::string& rule_id, RuleLifetime& lifetime,
+    SessionStateUpdateCriteria* update_criteria) {
+  RuleToProcess to_process;
+  PolicyRule rule;
+  static_rules_.get_rule(rule_id, &rule);
+  rule_lifetimes_[rule_id] = lifetime;
+  active_static_rules_.push_back(rule_id);
+  if (update_criteria) {
+    update_criteria->static_rules_to_install.insert(rule_id);
+    update_criteria->new_rule_lifetimes[rule_id] = lifetime;
+  }
+  increment_rule_stats(rule_id, update_criteria);
+  to_process.version = get_current_rule_version(rule_id);
+  to_process.rule    = rule;
+  to_process.teids   = config_.common_context.teids();
+  return to_process;
+}
+
+optional<RuleToProcess> SessionState::deactivate_static_5g_rule(
+    const std::string& rule_id, SessionStateUpdateCriteria* update_criteria) {
+  RuleToProcess to_process;
+  auto it = std::find(
+      active_static_rules_.begin(), active_static_rules_.end(), rule_id);
+  if (it == active_static_rules_.end()) {
+    return {};
+  }
+  update_criteria->static_rules_to_uninstall.insert(rule_id);
+  active_static_rules_.erase(it);
+  increment_rule_stats(rule_id, update_criteria);
+
+  PolicyRule rule;
+  if (!static_rules_.get_rule(rule_id, &rule)) {
+    rule.set_id(rule_id);
+  }
+
+  to_process.version = get_current_rule_version(rule.id());
+  to_process.rule    = rule;
+  to_process.teids   = config_.common_context.teids();
+  return to_process;
+}
+
+RuleToProcess SessionState::insert_dynamic_5g_rule(
+    const PolicyRule& rule, const RuleLifetime& lifetime,
+    SessionStateUpdateCriteria* session_uc) {
+  RuleToProcess to_process;
+  rule_lifetimes_[rule.id()] = lifetime;
+  dynamic_rules_.insert_rule(rule);
+  if (session_uc) {
+    session_uc->dynamic_rules_to_install.push_back(rule);
+    session_uc->new_rule_lifetimes[rule.id()] = lifetime;
+  }
+  increment_rule_stats(rule.id(), session_uc);
+
+  to_process.version = get_current_rule_version(rule.id());
+  to_process.rule    = rule;
+  to_process.teids   = config_.common_context.teids();
+
+  return to_process;
+}
+
+optional<RuleToProcess> SessionState::remove_dynamic_5g_rule(
+    const std::string& rule_id, PolicyRule* rule_out,
+    SessionStateUpdateCriteria* session_uc) {
+  PolicyRule rule;
+  RuleToProcess to_process;
+  bool removed = dynamic_rules_.remove_rule(rule_id, &rule);
+  if (!removed) {
+    return {};
+  }
+  if (rule_out) {
+    *rule_out = rule;
+  }
+  if (session_uc) {
+    session_uc->dynamic_rules_to_uninstall.insert(rule_id);
+  }
+  increment_rule_stats(rule_id, session_uc);
+
+  to_process.version = get_current_rule_version(rule.id());
+  to_process.rule    = rule;
+  to_process.teids   = config_.common_context.teids();
+
+  return to_process;
+}
+
+void SessionState::process_get_5g_rule_installs(
```

Comment:
why do we need `process_get_5g_rule_installs` 

Why can't we use the 4g path `process_get_rule_installs`

Reply:
process_get_rule_installs is included the bearer policy and scheduling policy. These are not supported in 5G.

Reply:
can you define what `process_get_5g_rule_installs` . It is confusing to me by its name
Maybe look for a more descriptive name